framework/include/bcs/VectorDivPenaltyDirichletBC.h

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "VectorIntegratedBC.h"
+
+/**
+ *  Enforces a Dirichlet boundary condition for the divergence of vector nonlinear
+ *  variables in a weak sense by applying a penalty to the difference between the
+ *  current solution and the Dirichlet data.
+ */
+class VectorDivPenaltyDirichletBC : public VectorIntegratedBC
+{
+public:
+  static InputParameters validParams();
+
+  VectorDivPenaltyDirichletBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual();
+  virtual Real computeQpJacobian();
```

Comment:
```suggestion
  virtual Real computeQpResidual() override;
  virtual Real computeQpJacobian() override;
```

Reply:
Done.

Reply:
I was about to suggest we should add a clang-tidy test, which would've caught this, https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-override.html.
But it seems work might've already started in #23614?

Reply:
We'd have to ask @loganharbour the status of that. I'm honestly surprised our `-Wall` recipes didn't catch this (or maybe CI just didn't have enough time to get there)

Reply:
Nah, there are multiple other occurrences throughout the code, the most notable in `VectorCurlPenaltyDirichletBC.h` which we based our implementation on. :P

Reply:
I've added a commit to add the keyword to the headers in `framework/include/bcs` which missed it.